### ISSUE TITLE
Unsupported cast from I64 to Bool

### ISSUE THREAD

**Issue by bio43** [snipsco/tract#274]
Hi, I hit this error today:
```
[2020-05-26T13:09:42.326524742Z DEBUG tract_hir::infer::analyser] Starting step for #6 "Cast_5" Cast
[2020-05-26T13:09:42.326527855Z TRACE tract_hir::infer::analyser]   Input  #0: xI64 1
[2020-05-26T13:09:42.326529379Z TRACE tract_hir::infer::analyser]   Output #0: ..x?
[2020-05-26T13:09:42.326535128Z TRACE tract_hir::infer::rules] Building rules for ElementWiseOp(Cast { to: Bool })
[2020-05-26T13:09:42.326538361Z TRACE tract_hir::infer::rules] Applying rules for ElementWiseOp(Cast { to: Bool })
[2020-05-26T13:09:42.326540004Z TRACE tract_hir::infer::rules::solver]   Applying rule GivenRule { inputs[0].datum_type }
[2020-05-26T13:09:42.326542598Z TRACE tract_hir::infer::rules::solver]     Given rule: inputs[0].datum_type is I64
[2020-05-26T13:09:42.326544481Z TRACE tract_hir::infer::rules::solver]   Applying rule inputs[0].shape == outputs[0].shape
[2020-05-26T13:09:42.326548508Z TRACE tract_hir::infer::rules::solver]   Applying all rules
[2020-05-26T13:09:42.326550492Z TRACE tract_hir::infer::rules::solver]   Applying rule outputs[0].datum_type == Bool
[2020-05-26T13:09:42.326552332Z TRACE tract_hir::infer::rules::solver]   Applying all rules
[2020-05-26T13:09:42.326553368Z TRACE tract_hir::infer::rules::solver]   Applying all rules
[2020-05-26T13:09:42.326554314Z TRACE tract_hir::infer::rules::solver]   Solver exiting Context { inputs: [xI64 1], outputs: [Bool] }
[2020-05-26T13:09:42.326558449Z TRACE tract_hir::infer::rules] Solver done
[2020-05-26T13:09:42.326570174Z DEBUG tract_hir::infer::analyser] TractError(Msg("Failed analyse for node #6 \"Cast_5\" Cast"), State { next_error: Some(TractError(Msg("Eager eval"), State { next_error: Some(TractError(Msg("Unsupported cast from I64 to Bool"), State { next_error: None, backtrace: InternalBacktrace { backtrace: None } })), backtrace: InternalBacktrace { backtrace: None } })), backtrace: InternalBacktrace { backtrace: None } })

```

Would it be possible to support casting from I64 to bool? PyTorch will happily export the operation (calling .bool() on a LongTensor), so I assume it's valid ONNX.

**Comment by kali** [comment]
Hey @bio43 !

Could you give a shot at #276 ? this case is not covered by onnx test suite, so I only have a very tiny  partial test case, so please conform it's fixing it for you.

**Comment by bio43** [comment]
Hey @kali , thanks for a quick fix! I made a comment on your PR

**Comment by bio43** [comment]
Closing this as fixed in #276
